mbed/mbed.py

```diff
@@ -2651,10 +2694,28 @@ def test_(toolchain=None, target=None, compile_list=False, run_list=False, compi
                   + (['-v'] if verbose else [])
                   + (['--app-config', app_config] if app_config else [])
                   + (['--test-config', test_config] if test_config else [])
+                  + (['--greentea'] if icetea_supported and greentea else [])
                   + args,
                   env=env)
 
+        if compile_list and icetea:
+            popen(icetea_command_base + ['--compile-list'])
+
         if compile_only or build_and_run_tests:
+
+            # Add icetea binaries in compile list
+            tests_by_name_temp = tests_by_name if tests_by_name else ''
+            if icetea:
+                proc = popen(icetea_command_base + ['--application-list'], stdin=subprocess.PIPE, stdout=subprocess.PIPE,
+                             stderr=subprocess.STDOUT)
```

Comment:
Is the stdout output a comma separated string? Like `test1,test2,test3`? Just trying to understand this block.

Reply:
it is something like:
TEST_APPS-Something
TEST_APPS-Something_else